### PR TITLE
Added Sean's changes

### PR DESCRIPTION
**Pull request by maniyar1**

### CODE REVIEW

File: motornet/effector.py [hunk 1]
```diff
@@ -43,24 +44,26 @@ class Effector(th.nn.Module):
     
   def __init__(
     self,
-    skeleton,
-    muscle,
+    skeleton: Skeleton,
+    muscle: Muscle,
```
Comment (maniyar1) [inline motornet/effector.py:48]: Adding types for checking and docs

File: motornet/effector.py [hunk 2]
```diff
@@ -43,24 +44,26 @@ class Effector(th.nn.Module):
...
     damping: float = 0.,
-    pos_lower_bound: Union[float, list, tuple] = None,
-    pos_upper_bound: Union[float, list, tuple] = None,
-    vel_lower_bound: Union[float, list, tuple] = None,
-    vel_upper_bound: Union[float, list, tuple] = None,
+    pos_lower_bound: float | Sequence[float] | th.Tensor | None = None,
+    pos_upper_bound: float | Sequence[float] | th.Tensor | None = None,
+    vel_lower_bound: float | Sequence[float] | th.Tensor | None = None,
+    vel_upper_bound: float | Sequence[float] | th.Tensor | None = None,
+    device: th.device | None = None
```
Comment (maniyar1) [inline motornet/effector.py:58]: Use new py 3.10 Union syntax

File: motornet/effector.py [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
+from collections.abc import Sequence
 from gymnasium.utils import seeding
 from torch.nn.parameter import Parameter
-from motornet.skeleton import TwoDofArm, PointMass
-from motornet.muscle import CompliantTendonHillMuscle, ReluMuscle
+from motornet.skeleton import TwoDofArm, PointMass, Skeleton
+from motornet.muscle import CompliantTendonHillMuscle, ReluMuscle, Muscle
 
 
 DEVICE = th.device("cpu")
```
Comment (maniyar1) [inline motornet/effector.py:11]: Is this still needed?

File: motornet/effector.py [hunk 4]
```diff
@@ -43,24 +44,26 @@ class Effector(th.nn.Module):
...
+    vel_upper_bound: float | Sequence[float] | th.Tensor | None = None,
+    device: th.device | None = None
   ):
     
     super().__init__()
 
     self.__name__ = name
-    self.damping = Parameter(th.tensor(damping, dtype=th.float32), requires_grad=False)
-    self.skeleton = skeleton.to(self.device)
+    device = self.device if device is None else device
```
Comment (maniyar1) [inline motornet/effector.py:64]: how does this work?

File: motornet/effector.py [hunk 5]
```diff
@@ -70,43 +73,22 @@ def __init__(
...
-    self.vel_range_bound = Parameter(vel_range, requires_grad=False)
-
+    # build skeleton
     self.skeleton.build(
       timestep=self.dt,
-      pos_upper_bound=self.pos_upper_bound,
-      pos_lower_bound=self.pos_lower_bound,
-      vel_upper_bound=self.vel_upper_bound,
-      vel_lower_bound=self.vel_lower_bound,
+      pos_upper_bound=pos_upper_bound,
```
Comment (maniyar1) [inline motornet/effector.py:81]: what if these are None?

File: motornet/effector.py [hunk 6]
```diff
@@ -70,43 +73,22 @@ def __init__(
...
+      pos_lower_bound=pos_lower_bound,
+      vel_upper_bound=vel_upper_bound,
+      vel_lower_bound=vel_lower_bound,
     )
-
+    
     # initialize muscle system
-    self.muscle = muscle.to(self.device)
-    self.force_index = self.muscle.state_name.index('force')  # column index of muscle state containing output force
-    self.MusclePaths = []  # a list of all the muscle paths
```
Comment (maniyar1) [inline motornet/effector.py:104]: unsure what this was for in motornet to begin with

File: motornet/effector.py [hunk 7]
```diff
@@ -173,41 +146,48 @@ def reset(self, *, seed: int | None = None, options: dict[str, Any] | None = Non
...
     options = {} if options is None else options
     batch_size: int = options.get('batch_size', 1)
-    joint_state: th.Tensor | np.ndarray | None = options.get('joint_state', None)
+    joint_state: Sequence | th.Tensor | None = options.get('joint_state', None)
 
     if joint_state is not None:
-      joint_state_shape = np.shape(joint_state.cpu().detach().numpy())
-      if joint_state_shape[0] > 1:
-        batch_size = joint_state_shape[0]
+      joint_state = th.as_tensor(joint_state, dtype=th.float32, device=self.device)
```
Comment (slocke3) [inline motornet/effector.py:156]: Why should joint_state be 1 or 2?

File: motornet/effector.py [hunk 8]
```diff
@@ -173,41 +146,48 @@ def reset(self, *, seed: int | None = None, options: dict[str, Any] | None = Non
...
+      if joint_state.ndim != 1 and joint_state.ndim != 2:
```
Comment (slocke3) [inline motornet/effector.py:157]: Why should joint_state be 1 or 2?

File: motornet/effector.py [hunk 9]
```diff
@@ -173,41 +146,48 @@ def reset(self, *, seed: int | None = None, options: dict[str, Any] | None = Non
...
+        raise ValueError(f'`joint_state` should have 1 or 2 dimensions but has {joint_state.ndim}.')
+      batch_size = joint_state.shape[0] if joint_state.ndim == 2 and joint_state.shape[0] > 1 else batch_size
```
Comment (slocke3) [inline motornet/effector.py:159]: Why is this named batch_size?

Comment (maniyar1) [inline motornet/effector.py:159]: we dont need to check if it is equal to 2, thats guarenteed

File: motornet/environment.py [hunk 10]
```diff
@@ -352,25 +330,25 @@ def np_random(self) -> np.random.Generator:
...
 
-  def apply_noise(self, loc, noise: float | list) -> th.Tensor:
+  def apply_noise(self, loc: th.Tensor, scale: float | Sequence[float]) -> th.Tensor:
     """Applies element-wise Gaussian noise to the input `loc`.
 
     Args:
       loc: input on which the Gaussian noise is applied, which in probabilistic terms make it the mean of the
         Gaussian distribution.
-      noise: `Float` or `list`, the standard deviation (spread or “width”) of the distribution. Must be 
+      scale: `Float` or `list`, the standard deviation (spread or “width”) of the distribution. Must be
```
Comment (slocke3) [inline motornet/environment.py:342]: Why rename noise to scale?

**Pull request merged**